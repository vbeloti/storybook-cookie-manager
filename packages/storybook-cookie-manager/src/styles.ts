import { styled } from 'storybook/theming'
import { Form, Table as SBTable } from 'storybook/internal/components'

export const Content = styled.div`
  padding: 16px;
`

export const TableWrapper = styled.div`
  margin-bottom: 16px;
`

export const ButtonWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  margin-top: 16px;
`

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const Label: React.ComponentType<any> = styled(Form.Field)`
  margin: 0;
`

export const Input = styled(Form.Input)`
  width: 100%;
`

export const Table = styled(SBTable)`
  width: 100%;
`
