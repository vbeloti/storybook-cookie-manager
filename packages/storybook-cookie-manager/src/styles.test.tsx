import React from 'react'
import { render } from '@testing-library/react'
import 'jest-styled-components'
import { Content, TableWrapper, ButtonWrapper, Label, Input, Table } from './styles'

jest.mock('storybook/internal/components', () => ({
  Form: {
    Field: ({ children }: { children: React.ReactNode }) => <form>{children}</form>,
    Input: ({ children }: { children: React.ReactNode }) => <input>{children}</input>,
  },
  Table: ({ children }: { children: React.ReactNode }) => <table>{children}</table>,
  Button: ({ children, ...props }: { children: React.ReactNode }) => <button {...props}>{children}</button>,
}))

describe('styles', () => {
  it('should render Content correctly', () => {
    const { container } = render(<Content />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render TableWrapper correctly', () => {
    const { container } = render(<TableWrapper />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render ButtonWrapper correctly', () => {
    const { container } = render(<ButtonWrapper />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render Label correctly', () => {
    const { container } = render(<Label label="test" />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render Input correctly', () => {
    const { container } = render(<Input />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render Table correctly', () => {
    const { container } = render(
      <Table>
        <tbody>
          <tr>
            <td>test</td>
          </tr>
        </tbody>
      </Table>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
