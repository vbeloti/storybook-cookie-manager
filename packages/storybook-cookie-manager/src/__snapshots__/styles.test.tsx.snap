// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`styles should render ButtonWrapper correctly 1`] = `
<div
  class="css-ya232j"
/>
`;

exports[`styles should render Content correctly 1`] = `
<div
  class="css-152opwo"
/>
`;

exports[`styles should render Input correctly 1`] = `<input />`;

exports[`styles should render Label correctly 1`] = `<form />`;

exports[`styles should render Table correctly 1`] = `
<table>
  <tbody>
    <tr>
      <td>
        test
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`styles should render TableWrapper correctly 1`] = `
<div
  class="css-haw22i"
/>
`;
